app/services/medical_professionals/create_service.rb

```diff
@@ -0,0 +1,55 @@
+class MedicalProfessionals::CreateService
+  class << self
+    def create_doctor(doctor_params, insurances, specialties, profession)
+      new_doctor = Doctor.new(doctor_params)
+      if new_doctor.save
+        if insurances && specialties
+          add_insurance(new_doctor, insurances, profession)
+          add_specialties(new_doctor, specialties, profession)
+        elsif insurances
+          add_insurance(new_doctor, insurances, profession)
+        elsif specialties
```

Comment:
I think FE wants insurance to be required for doctors. So we would not need to cover this scenario.

Reply:
Ok cool. Would specialties be optional?

Reply:
Not sure, maybe ask at standup